sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/CreateJacksonParser.scala

```diff
@@ -55,19 +52,13 @@ private[sql] object CreateJacksonParser extends Serializable {
   //    automatically by analyzing first bytes of the input stream.
   // 3. Reader based parser. This is the slowest parser used here but it allows to create
   //    a reader with specific encoding.
-  // The method creates a reader for an array with given encoding and sets size of internal
-  // decoding buffer according to size of input array.
-  private def getStreamDecoder(enc: String, in: Array[Byte], length: Int): StreamDecoder = {
+  private def getReader(enc: String, in: Array[Byte], length: Int): Reader = {
     val bais = new ByteArrayInputStream(in, 0, length)
-    val byteChannel = Channels.newChannel(bais)
-    val decodingBufferSize = Math.min(length, 8192)
-    val decoder = Charset.forName(enc).newDecoder()
-
-    StreamDecoder.forDecoder(byteChannel, decoder, decodingBufferSize)
+    new InputStreamReader(bais, enc)
```

Comment:
Seems fine but I'm wondering why it used channels in the first place. Was it for performance?

Reply:
I took the code from internals of Jackson parser, and kept it as is to be sure that the behavior is the same. `InputStreamReader` uses similar code but without creating a channel. So, I think we could use the standard API of `InputStreamReader`.